percent_box.js

```diff
@@ -113,10 +113,19 @@ class PercentBoxElement extends HTMLElement {
       this._pbox.update(data);
       toggleSVGContainerAndIndicator.call(this, false);
       broker.addEventListener('stats-stream-data', (evt) => {
-        const stats = evt.detail;
-        const val = stats[this.percentKey];
-        const total = stats[this.totalKey];
-        data = [ val, total - val ];
+        const stats = evt.detail.stats;
+        const { mappedReads, unmappedReads } = evt.detail.mapDataFromIndex
+
+        if (this.percentKey === "mapped_reads" && mappedReads != undefined & unmappedReads != undefined) {
+          data = [mappedReads, unmappedReads]
+          this._pbox.chart.setColors('rgb(9,176,135)', 'rgba(9,176,135,0.5)');
```

Comment:
I don't think we need to make changes to iobio.viz. You can use JS to set the CSS custom properties we're using for colors:

```javascript
const pbox = document.querySelector('#mapped-reads-chart');
pbox.style.setProperty('--iobio-data-color', 'rgb(9,176,135)');
pbox.style.setProperty('--iobio-data-color-secondary', 'rgba(9,176,135,0.5)');
```

I'm not sure where the best place to put that is. Not in the data broker, but you need to detect whether the mapped reads data is available in the index before you call it. I'll leave that to you.